src/components/posts/List/List.Actions.js

```diff
@@ -1,13 +1,14 @@
+
 import React from 'react'
 import { Link } from 'react-router-dom'
 import moment from 'moment'
 
-export default ({ destroyPost, post, user }) => (
+ export default ({ destroyPost, post, user }) => (
   <div className='card-footer text-muted d-flex justify-content-around'>
     <Link className='btn btn-link' to={`/users/${user._id}/posts/${post._id}/edit`}>Edit Post</Link>
     <button
       className='btn btn-link text-danger'
-      onClick={() => destroyPost(post)}>
+      onClick={() => destroyPost(user, post)}>
```

Comment:
Looking at the destroy post function, you only need to pass the post. By passing the user first , the 
`destroyPost (post) {` function is interpreting the user object to be the post and is sending the wrong post id to the server for deletion!